Atmel Saml2x dts fix

jmalbers opened the pull request:
Updates Atmel SAML device tree files saml2x.dtsi and saml21.dtsi

Many devices do not share registers between saml21 and saml22 socs. 

All devices without shared addresses were moved into the saml21.dtsi

Incorrect power manager address was corrected.

Timer counter 0 - 3 were added to saml2x device tree include.

Built and ran blinky example and hello world.

Commit:
Move unshared device tree entries from saml2x.dtsi to saml21.dtsi

Atmel SAML21 and SAML22 devices share few device addresses.

All devices with addresses and configurations unique to the saml21 soc
have been moved out of the saml2x base dtsi file and into saml21.dtsi

Also:

- Change conflicting PM address register.

- Add missing timer counter tc0 to tc3 devices to saml2x device tree.

Signed-off-by: Jacob M. Albers <jmalbers@gmail.com>

stephanosio reviewed (changes requested):
Please review the [Contribution Guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines) and re-format the commit message.

jmalbers commented:
Thanks for the help. First timer here :)

nandojve commented:
Hi @jmalbers ,

Thank you for look on this. I noted that there are peripherals that you are moving which has same register address. The devicetree should keep shared address at only one file and describe the difference in the more specific file. Try follow [samv71q21b.dtsi](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/atmel/samv71q21b.dtsi) as referece.
You need structure the change using same principle.

jmalbers commented:
> Hi @jmalbers ,
> 
> Thank you for look on this. I noted that there are peripherals that you are moving which has same register address. The devicetree should keep shared address at only one file and describe the difference in the more specific file. Try follow [samv71q21b.dtsi](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/atmel/samv71q21b.dtsi) as referece. You need structure the change using same principle.

Missed a few I see! Thanks for pointing this out :)

nandojve commented on dts/arm/atmel/saml2x.dtsi:
```diff
@@ -45,160 +37,41 @@
-			interrupts = <27 0>;
-		};
 	};
+
```
This line is not necessary

nandojve commented on dts/arm/atmel/saml21.dtsi:
```diff
@@ -7,29 +7,205 @@
-			interrupts = <6 0>;
-			num-bidir-endpoints = <8>;
+
+		nvmctrl: nvmctrl@41004000  {
```
nvmctrl@41004000 is the same for L21/22. There is a big error at Figure 8-1. SAM L22 Product Mapping datasheet on the AHB-APB Bridge B part. The prefix of AHB-APB Bridge B is 0x41xx-xxxx and there are entries with 0x40xx-xxxx which is complete wrong.

nandojve commented on dts/arm/atmel/saml21.dtsi:
```diff
@@ -7,29 +7,205 @@
+
+			flash0: flash@0 {
+				compatible = "soc-nv-flash";
+				reg = <0 0x40000>;
```
flash@0 reg property should be defined on a file named saml21x18.dtsi
idem with sram0: memory@20000000, the reg property should be at saml21x18.dtsi file too.

Same principle of https://github.com/zephyrproject-rtos/zephyr/blob/32f4b6e09a85c4886767df93d8a6c4297b79b009/dts/arm/atmel/samd5xx20.dtsi#L7-L22

nandojve commented:
Hi @jmalbers ,

I was wondering if you had time to look comments. I would like to have it for next release but a few work still needed. Please, let us know if you still have interest and time to finish work.

jmalbers commented:
Sorry, let this sit for a while. I'll have a look at it again this week.

jmalbers committed:
Add saml21x18.dtsi, cleanup associated dts

Created saml21x18.dtsi for sram and flash reg / size entries

Moved nvmctrl dts entry to saml2x.dtsi

Changed saml21_xpro.dts to include saml21x18.dtsi instead of saml21.dtsi

Deleted unnecessary spaces

Signed-off-by: Jacob M. Albers <jmalbers@gmail.com>

jmalbers committed:
Merge branch 'main' into samlxx_dts_fix

nandojve commented:
Hi @jmalbers ,

You need rebase and keep your commits on top of main without merge commits.
May you update?

nandojve commented:
Hi @jmalbers ,

Merge windows is open again and we can continue work here.
